server.py

```diff
@@ -1,75 +1,134 @@
-import settings
-import json
-import html
-import logging
-import logging.handlers
-import requests
-from flask import Flask, request, Response, jsonify, abort
+from flask import Flask, request, jsonify
 from pymongo import MongoClient
 import pymongo.errors
 from datetime import datetime
 from voluptuous import Schema, Coerce, All, Range, MultipleInvalid
 from bson.objectid import ObjectId
 from bson.errors import InvalidId
 import pika
+import settings
+import logging
+from structlog import wrap_logger
 
+logger = wrap_logger(logging.getLogger(__name__))
 app = Flask(__name__)
-
 app.config['MONGODB_URL'] = settings.MONGODB_URL
-logger = settings.logger
-
-mongo_client = MongoClient(app.config['MONGODB_URL'])
-db = mongo_client.sdx_store
-
-def queue_notification(notification):
-    logger.debug(" [x] Queuing notification to " + settings.RABBIT_QUEUE)
-    logger.debug(notification)
-    connection = pika.BlockingConnection(pika.URLParameters(settings.RABBIT_URL))
-    channel = connection.channel()
-    channel.queue_declare(queue=settings.RABBIT_QUEUE)
-    channel.basic_publish(exchange='',
-                          routing_key=settings.RABBIT_QUEUE,
-                          body=notification)
-    logger.debug(" [x] Queued notification to " + settings.RABBIT_QUEUE)
-    connection.close()
 
-@app.route('/responses', methods=['POST'])
-def do_save_response():
-    survey_response = request.get_json(force=True)
+schema = Schema({
+    'survey_id': str,
+    'form': str,
+    'ru_ref': str,
+    'period': str,
+    'added_ms': Coerce(int),
+    'per_page': All(Coerce(int), Range(min=1, max=100)),
+    'page': All(Coerce(int), Range(min=1))
+})
+
+
+def get_db_responses():
+    mongo_client = MongoClient(app.config['MONGODB_URL'])
+    return mongo_client.sdx_store.responses
+
+
+@app.errorhandler(400)
+def errorhandler_400(e):
+    return client_error(repr(e))
+
+
+def client_error(error=None):
+    logger.error(error, status=400)
+    message = {
+        'status': 400,
+        'message': error,
+        'uri': request.url
+    }
+    resp = jsonify(message)
+    resp.status_code = 400
+
+    return resp
+
+
+@app.errorhandler(500)
+def errorhandler_500(e):
+    return server_error(repr(e))
+
+
+@app.errorhandler(500)
+def server_error(error=None):
+    logger.error(error, status=500)
+    message = {
+        'status': 500,
+        'message': error
+    }
+    resp = jsonify(message)
+    resp.status_code = 500
+
+    return resp
+
+
+def queue_notification(notification, bound_logger):
+    bound_logger.debug(" [x] Queuing notification to " + settings.RABBIT_QUEUE)
```

Comment:
We should try and do this with structured logging e.g. `bound_logger.debug("Queue notification", queue=settings.RABBIT_QUEUE)`